File: src/Application Specific/Classes/Entity Collections/Jobs/JobItem/JobItemSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace JobItems

        Public Class JobItemSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Sub Delete(ByVal JobItemIDs As String, ByVal JobOfWorkID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@JobItemIDs", JobItemIDs, True)
                _database.AddParameter("@JobOfWorkID", JobOfWorkID, True)

                _database.ExecuteSP_NO_Return("JobItem_Delete")
            End Sub

            Public Sub DeleteAll_ForJOW(ByVal JobOfWorkID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@JobOfWorkID", JobOfWorkID, True)
                _database.ExecuteSP_NO_Return("JobItem_DeleteALL_ForJOW")
            End Sub

            Public Sub Delete(ByVal JobItemIDs As String, ByVal JobOfWorkID As Integer, ByVal trans As SqlClient.SqlTransaction)

                Dim Command As SqlClient.SqlCommand = New SqlClient.SqlCommand

                Command.CommandText = "JobItem_Delete"
                Command.CommandType = CommandType.StoredProcedure
                Command.Transaction = trans
                Command.Connection = trans.Connection

                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@JobOfWorkID", JobOfWorkID))
                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@JobItemIDs", JobItemIDs))
                Command.ExecuteNonQuery()

            End Sub

            Public Function [Insert](ByVal oJobItem As JobItem, ByVal trans As SqlClient.SqlTransaction) As JobItem

                Dim Command As SqlClient.SqlCommand = New SqlClient.SqlCommand

                Dim IDs As String = ""
                'AMY CHANGED THIS BECAUSE I NEED TO RETAIN THE LINK!
                If oJobItem.JobItemID > 0 Then
                    'UPDATE
                    Command.CommandText = "JobItem_Update"
                    Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@JobItemID", oJobItem.JobItemID))
                Else
                    'INSERT
                    Command.CommandText = "JobItem_Insert"
                End If

                Command.CommandType = CommandType.StoredProcedure
                Command.Transaction = trans
                Command.Connection = trans.Connection

                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@JobOfWorkID", oJobItem.JobOfWorkID))
                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@Summary", oJobItem.Summary))
                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@RateID", oJobItem.RateID))
                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@Qty", oJobItem.Qty))
                Command.Parameters.Add(New System.Data.SqlClient.SqlParameter("@SystemLinkID", oJobItem.SystemLinkID))

                If oJobItem.JobItemID > 0 Then
                    Command.ExecuteNonQuery()
                Else
                    oJobItem.SetJobItemID = Command.ExecuteScalar
                End If

                Return oJobItem
            End Function

            Public Function [Insert](ByVal oJobItem As JobItem) As JobItem
                _database.ClearParameter()
                _database.AddParameter("@JobOfWorkID", oJobItem.JobOfWorkID, True)
                _database.AddParameter("@Summary", oJobItem.Summary, True)
                _database.AddParameter("@RateID", oJobItem.RateID, True)
                _database.AddParameter("@Qty", oJobItem.Qty, True)
                _database.AddParameter("@SystemLinkID", oJobItem.SystemLinkID, True)

                Dim IDs As String = ""
                'AMY CHANGED THIS BECAUSE I NEED TO RETAIN THE LINK!
                If oJobItem.JobItemID > 0 Then
                    'UPDATE
                    _database.AddParameter("@JobItemID", oJobItem.JobItemID, True)
                    _database.ExecuteSP_NO_Return("JobItem_Update")
                Else
                    'INSERT
                    oJobItem.SetJobItemID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("JobItem_Insert"))
                End If

                Return oJobItem
            End Function

            Public Function JobItems_Get_For_Job_Of_Work(ByVal JobOfWorkID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@JobOfWorkID", JobOfWorkID, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("JobItems_Get_For_Job_Of_Work")
                dt.TableName = Entity.Sys.Enums.TableNames.tblJobItem.ToString
                Return New DataView(dt)
            End Function

            Public Function JobItems_Get_For_Job_Of_Work(ByVal JobOfWorkID As Integer, ByVal trans As SqlClient.SqlTransaction) As DataView

                Dim Command As SqlClient.SqlCommand = New SqlClient.SqlCommand

                Command.CommandText = "JobItems_Get_For_Job_Of_Work"
                Command.CommandType = CommandType.StoredProcedure
                Command.Transaction = trans
                Command.Connection = trans.Connection

                Command.Parameters.AddWithValue("@JobOfWorkID", JobOfWorkID)

                Dim Adapter As New SqlClient.SqlDataAdapter(Command)
                Dim returnDS As New DataSet
                Adapter.Fill(returnDS)

                Dim dt As DataTable = returnDS.Tables(0)

                dt.TableName = Entity.Sys.Enums.TableNames.tblJobItem.ToString
                Return New DataView(dt)
            End Function

            Public Function JobOfWork_Get_For_Job_Of_Work_As_Objects(ByVal JobOfWorkID As Integer) As ArrayList
                Dim jobItems As New ArrayList
                For Each row As DataRow In JobItems_Get_For_Job_Of_Work(JobOfWorkID).Table.Rows
                    Dim jobItem As New JobItems.JobItem
                    jobItem.IgnoreExceptionsOnSetMethods = True
                    jobItem.SetJobItemID = row.Item("JobItemID")
                    jobItem.SetJobOfWorkID = row.Item("JobOfWorkID")
                    jobItem.SetSummary = row.Item("Summary")
                    jobItem.SetRateID = row.Item("RateID")
                    jobItem.SetQty = row.Item("Qty")
                    If row.Table.Columns.Contains("SystemLinkID") Then jobItem.SetSystemLinkID = row.Item("SystemLinkID")
                    jobItems.Add(jobItem)
                Next

                Return jobItems
            End Function

            Public Function JobOfWork_Get_For_Job_Of_Work_As_Objects(ByVal JobOfWorkID As Integer, ByVal trans As SqlClient.SqlTransaction) As ArrayList
                Dim jobItems As New ArrayList
                For Each row As DataRow In JobItems_Get_For_Job_Of_Work(JobOfWorkID, trans).Table.Rows
                    Dim jobItem As New JobItems.JobItem
                    jobItem.IgnoreExceptionsOnSetMethods = True
                    jobItem.SetJobItemID = row.Item("JobItemID")
                    jobItem.SetJobOfWorkID = row.Item("JobOfWorkID")
                    jobItem.SetSummary = row.Item("Summary")
                    jobItem.SetRateID = row.Item("RateID")
                    jobItem.SetQty = row.Item("Qty")
                    If row.Table.Columns.Contains("SystemLinkID") Then jobItem.SetSystemLinkID = row.Item("SystemLinkID")
                    jobItems.Add(jobItem)
                Next

                Return jobItems
            End Function

            Public Function JobItems_Get_For_Job(ByVal JobID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@JobID", JobID, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("JobItems_Get_For_Job")
                dt.TableName = Entity.Sys.Enums.TableNames.tblJobItem.ToString
                Return New DataView(dt)
            End Function

#End Region

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Jobs/JobItem/JobItemSQL.cs
using System.Collections;
using System.Data;
using System.Data.SqlClient;

namespace FSM.Entity
{
    namespace JobItems
    {
        public class JobItemSQL
        {
            private Sys.Database _database;

            public JobItemSQL(Sys.Database database)
            {
                _database = database;
            }

            
            public void Delete(string JobItemIDs, int JobOfWorkID)
            {
                _database.ClearParameter();
                _database.AddParameter("@JobItemIDs", JobItemIDs, true);
                _database.AddParameter("@JobOfWorkID", JobOfWorkID, true);
                _database.ExecuteSP_NO_Return("JobItem_Delete");
            }

            public void DeleteAll_ForJOW(int JobOfWorkID)
            {
                _database.ClearParameter();
                _database.AddParameter("@JobOfWorkID", JobOfWorkID, true);
                _database.ExecuteSP_NO_Return("JobItem_DeleteALL_ForJOW");
            }

            public void Delete(string JobItemIDs, int JobOfWorkID, SqlTransaction trans)
            {
                var Command = new SqlCommand();
                Command.CommandText = "JobItem_Delete";
                Command.CommandType = CommandType.StoredProcedure;
                Command.Transaction = trans;
                Command.Connection = trans.Connection;
                Command.Parameters.Add(new SqlParameter("@JobOfWorkID", JobOfWorkID));
                Command.Parameters.Add(new SqlParameter("@JobItemIDs", JobItemIDs));
                Command.ExecuteNonQuery();
            }

            public JobItem Insert(JobItem oJobItem, SqlTransaction trans)
            {
                var Command = new SqlCommand();
                string IDs = "";
                // AMY CHANGED THIS BECAUSE I NEED TO RETAIN THE LINK!
                if (oJobItem.JobItemID > 0)
                {
                    // UPDATE
                    Command.CommandText = "JobItem_Update";
                    Command.Parameters.Add(new SqlParameter("@JobItemID", oJobItem.JobItemID));
                }
                else
                {
                    // INSERT
                    Command.CommandText = "JobItem_Insert";
                }

                Command.CommandType = CommandType.StoredProcedure;
                Command.Transaction = trans;
                Command.Connection = trans.Connection;
                Command.Parameters.Add(new SqlParameter("@JobOfWorkID", oJobItem.JobOfWorkID));
                Command.Parameters.Add(new SqlParameter("@Summary", oJobItem.Summary));
                Command.Parameters.Add(new SqlParameter("@RateID", oJobItem.RateID));
                Command.Parameters.Add(new SqlParameter("@Qty", oJobItem.Qty));
                Command.Parameters.Add(new SqlParameter("@SystemLinkID", oJobItem.SystemLinkID));
                if (oJobItem.JobItemID > 0)
                {
                    Command.ExecuteNonQuery();
                }
                else
                {
                    oJobItem.SetJobItemID = Command.ExecuteScalar();
                }

                return oJobItem;
            }

            public JobItem Insert(JobItem oJobItem)
            {
                _database.ClearParameter();
                _database.AddParameter("@JobOfWorkID", oJobItem.JobOfWorkID, true);
                _database.AddParameter("@Summary", oJobItem.Summary, true);
                _database.AddParameter("@RateID", oJobItem.RateID, true);
                _database.AddParameter("@Qty", oJobItem.Qty, true);
                _database.AddParameter("@SystemLinkID", oJobItem.SystemLinkID, true);
                string IDs = "";
                // AMY CHANGED THIS BECAUSE I NEED TO RETAIN THE LINK!
                if (oJobItem.JobItemID > 0)
                {
                    // UPDATE
                    _database.AddParameter("@JobItemID", oJobItem.JobItemID, true);
                    _database.ExecuteSP_NO_Return("JobItem_Update");
                }
                else
                {
                    // INSERT
                    oJobItem.SetJobItemID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("JobItem_Insert"));
                }

                return oJobItem;
            }

            public DataView JobItems_Get_For_Job_Of_Work(int JobOfWorkID)
            {
                _database.ClearParameter();
                _database.AddParameter("@JobOfWorkID", JobOfWorkID, true);
                var dt = _database.ExecuteSP_DataTable("JobItems_Get_For_Job_Of_Work");
                dt.TableName = Sys.Enums.TableNames.tblJobItem.ToString();
                return new DataView(dt);
            }

            public DataView JobItems_Get_For_Job_Of_Work(int JobOfWorkID, SqlTransaction trans)
            {
                var Command = new SqlCommand();
                Command.CommandText = "JobItems_Get_For_Job_Of_Work";
                Command.CommandType = CommandType.StoredProcedure;
                Command.Transaction = trans;
                Command.Connection = trans.Connection;
                Command.Parameters.AddWithValue("@JobOfWorkID", JobOfWorkID);
                var Adapter = new SqlDataAdapter(Command);
                var returnDS = new DataSet();
                Adapter.Fill(returnDS);
                var dt = returnDS.Tables[0];
                dt.TableName = Sys.Enums.TableNames.tblJobItem.ToString();
                return new DataView(dt);
            }

            public ArrayList JobOfWork_Get_For_Job_Of_Work_As_Objects(int JobOfWorkID)
            {
                var jobItems = new ArrayList();
                foreach (DataRow row in JobItems_Get_For_Job_Of_Work(JobOfWorkID).Table.Rows)
                {
                    var jobItem = new JobItem();
                    jobItem.IgnoreExceptionsOnSetMethods = true;
                    jobItem.SetJobItemID = row["JobItemID"];
                    jobItem.SetJobOfWorkID = row["JobOfWorkID"];
                    jobItem.SetSummary = row["Summary"];
                    jobItem.SetRateID = row["RateID"];
                    jobItem.SetQty = row["Qty"];
                    if (row.Table.Columns.Contains("SystemLinkID"))
                        jobItem.SetSystemLinkID = row["SystemLinkID"];
                    jobItems.Add(jobItem);
                }

                return jobItems;
            }

            public ArrayList JobOfWork_Get_For_Job_Of_Work_As_Objects(int JobOfWorkID, SqlTransaction trans)
            {
                var jobItems = new ArrayList();
                foreach (DataRow row in JobItems_Get_For_Job_Of_Work(JobOfWorkID, trans).Table.Rows)
                {
                    var jobItem = new JobItem();
                    jobItem.IgnoreExceptionsOnSetMethods = true;
                    jobItem.SetJobItemID = row["JobItemID"];
                    jobItem.SetJobOfWorkID = row["JobOfWorkID"];
                    jobItem.SetSummary = row["Summary"];
                    jobItem.SetRateID = row["RateID"];
                    jobItem.SetQty = row["Qty"];
                    if (row.Table.Columns.Contains("SystemLinkID"))
                        jobItem.SetSystemLinkID = row["SystemLinkID"];
                    jobItems.Add(jobItem);
                }

                return jobItems;
            }

            public DataView JobItems_Get_For_Job(int JobID)
            {
                _database.ClearParameter();
                _database.AddParameter("@JobID", JobID, true);
                var dt = _database.ExecuteSP_DataTable("JobItems_Get_For_Job");
                dt.TableName = Sys.Enums.TableNames.tblJobItem.ToString();
                return new DataView(dt);
            }

            
        }
    }
}